### PR TITLE
docs: update release process

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

Title says it all.

### Context of Change

#315 added integration tests to every PR, so that step is no longer needed as a part of the release process.

### Type of Change

- [x] Documentation Updates

## Test Plan

CI passes.

### CODE REVIEW

**Review by JST5000 — APPROVED** [review]

**Review by JST5000 — COMMENTED** [review]
Actually, looking at the GitHub actions it seems we only run the integration tests on the PR, not the master branch after it gets merged. (So there could be issues which don't exist on the PR branch, but do exist once it is merged - So we probably still need this step just to be safe, or we need to automatically run integration tests on the master branch after merging a PR)

File: CHANGELOG.md [hunk 1]
```diff
@@ -60,6 +59,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Makes the account delete fee dynamic, based on the ledger's reserve, instead of hard-coded
 - Fee scaling based on load on the ledger
 - Fixes potential issue with conflicting Decimal contexts
+- Fixes bug where autofilling using an `xAddress` with `None` in the Destination Tag threw a KeyError
```
Comment (ledhed2222) [inline CHANGELOG.md:62]: why move this?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -129,27 +129,25 @@ open _build/html/index.html
 
 * Your changes should have unit and/or integration tests.
 * Your changes should pass the linter.
-* Your code should pass all the unit tests on Github (which check all 3 versions of Python).
+* Your code should pass all the unit and integration tests on Github (which check all 4 versions of Python).
```
Comment (ledhed2222) [inline CONTRIBUTING.md:132]: can just remove the number to future proof this

**Review by ledhed2222 — APPROVED** [review]
a few comments but overall lgtm

**Review by ckniffen — APPROVED** [review]

**Review by khancode — APPROVED** [review]

**Pull request merged**